x/axelarnet/types/evm_interpreter.go

```diff
@@ -0,0 +1,111 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+var (
+	stringType  = funcs.Must(abi.NewType("string", "string", nil))
+	uint256Type = funcs.Must(abi.NewType("uint256", "uint256", nil))
+	bytesType   = funcs.Must(abi.NewType("bytes", "bytes", nil))
+
+	// abi encoded bytes, with the following format:
+	// wasm method name, argument number, comma seperated argument types, encoded bytes contain actual arguments name and value
+	payloadArguments = abi.Arguments{{Type: stringType}, {Type: uint256Type}, {Type: stringType}, {Type: bytesType}}
```

Comment:
I don't quite get it. Why not go with something like
```
abi.Arguments{{Type: "string"}, {Type: "[]string"}, {Type: "[]string"}, {Type: "[]bytes"}}
```
where the first "string" is method name, the second "[]string" is arg types, the third "[]string" is arg names, and the last "[]bytes" is arg values. You avoid using `,` as delimiter and also don't need `argNum`.